### PR TITLE
Remove base node related logics

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/3d-tasks/issues/9283 

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/scene-graph/scene.ts [hunk 1]
```diff
@@ -150,156 +133,28 @@ export class Scene extends BaseNode {
...
-     */
-    public get eulerAngles (): Readonly<Vec3> { return Vec3.ZERO; }
-
-    /**
-     * @en
-     * Refer to [[Node.worldMatrix]]
-     * @zh
-     * 参考 [[Node.worldMatrix]]
-     */
-    public get worldMatrix (): Readonly<Mat4> { return Mat4.IDENTITY; }
```
Comment (minggo) [inline cocos/core/scene-graph/scene.ts:301]: Why remove these functions?

Comment (novain1102) [reply cocos/core/scene-graph/scene.ts:301]: Now can directly use parent functions ([Node.worldMatrix]). And those Readonly functions will get errors when editor init

Comment (minggo) [reply cocos/core/scene-graph/scene.ts:301]: Got it. Is it related with this issue? I mean, can only delete these functions only remove BaseNode?

**Review by dumganhar — APPROVED** [review]

File: cocos/core/scene-graph/scene.ts (hunk 1 above)
Comment (novain1102) [reply cocos/core/scene-graph/scene.ts:301]: Yes. They are related.

**Review by minggo — APPROVED** [review]

**Comment by mmyduckx** [comment]
Arealy tested in latest 3.6.1. can merge now.

**Comment by novain1102** [comment]
Arealy tested in develop. can merge now.

**Comment by minggo** [comment]
@mmyduckx please fix ci error.

**Comment by novain1102** [comment]
> @mmyduckx please fix ci error.

fixed.

File: cocos/core/scene-graph/scene.ts [hunk 2]
```diff
@@ -150,156 +133,28 @@ export class Scene extends BaseNode {
...
+    public _onPostActivated (active: boolean) {
+
+    }
+
     /**
      * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
      */
     public _onBatchCreated (dontSyncChildPrefab: boolean) {
-        super._onBatchCreated(dontSyncChildPrefab);
+        if (this._parent) {
```
Comment (SantyWang) [inline cocos/core/scene-graph/scene.ts]: Seems redundant

File: cocos/core/scene-graph/scene.ts [hunk 3]
```diff
@@ -340,6 +195,14 @@ export class Scene extends BaseNode {
             this._globals.activate();
         }
     }
+
+    public _onSetParent (oldParent: this | null, keepWorldTransform = false) {
+        if (this._parent) {
```
Comment (SantyWang) [inline cocos/core/scene-graph/scene.ts]: Seems redundant

**Comment by SantyWang** [comment]
Search ` instanceof Node` globally, as it is now written in such a way that it can search for scene, which may cause problems

File: cocos/core/scene-graph/scene.ts (hunk 2 above)
Comment (novain1102) [reply cocos/core/scene-graph/scene.ts]: removed.

File: cocos/core/scene-graph/scene.ts (hunk 3 above)
Comment (novain1102) [reply cocos/core/scene-graph/scene.ts]: removed.

**Comment by novain1102** [comment]
> Search ` instanceof Node` globally, as it is now written in such a way that it can search for scene, which may cause problems

Already check.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**